lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -230,15 +259,15 @@ namespace Js
             OUTPUT_TRACE_DEBUGONLY(Js::ModulePhase, _u("\t>NotifyParentsAsNeeded\n"));
             NotifyParentsAsNeeded();
 
-            if (!WasDeclarationInitialized() && isRootModule)
+            if (this->promiseCapability != nullptr || (!WasDeclarationInitialized() && isRootModule))
             {
                 // TODO: move this as a promise call? if parser is called from a different thread
```

Comment:
This PR doesn't enable the case where parser is called from a different thread. Therefore I leave this TODO here for future enhancement.

---
In reply to: [114906209](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r114906209) [](ancestors = 114906209)